monkestation/code/controllers/subsystem/plexora.dm

```diff
@@ -0,0 +1,370 @@
+	var/target_ckey = input["target"]
+	var/selected_smite = input["smite"]
+	var/smited_by = input["smited_by_ckey"]
+
+	if (!GLOB.smites[selected_smite])
+		return "error=invalidsmite"
+
+	// DIVINE SMITING!
+	var/client/client = disambiguate_client(target_ckey)
+
+	if (!client)
+		return "error=clientnotexist"
+
+	var/smite_path = GLOB.smites[selected_smite]
+	var/datum/smite/picking_smite = new smite_path
+	var/configuration_success = picking_smite.configure(client)
+	if (configuration_success == FALSE)
+		return
+
+	// Mock admin
+	var/datum/client_interface/mockadmin = new(
+		key = smited_by,
+	)
+
+	src = mockadmin
+	picking_smite.effect(client, client.mob)
+
+/datum/world_topic/plx_ticketaction
+	keyword = "PLX_ticketaction"
+	require_comms_key = TRUE
+
+/datum/world_topic/plx_ticketaction/Run(list/input)
+	var/ticketid = text2num(input["id"])
+	var/action_by_ckey = input["action_by"]
+	var/action = input["action"]
+
+
+	var/datum/admin_help/ticket = GLOB.ahelp_tickets.TicketByID(ticketid)
+	if (!ticket) return "error=couldntfetchticket"
+
+	if (action != "reopen" && ticket.state != AHELP_ACTIVE)
+		return
+
+	switch(action)
+		if("reopen")
+			if (ticket.state == AHELP_ACTIVE) return
+			SSplexora.aticket_reopened(ticket, action_by_ckey)
+			ticket.Reopen(action_by_ckey)
+		if("reject")
+			SSplexora.aticket_closed(ticket, action_by_ckey, AHELP_CLOSETYPE_REJECT)
+			ticket.Reject(action_by_ckey)
+		if("icissue")
+			SSplexora.aticket_closed(ticket, action_by_ckey, AHELP_CLOSETYPE_RESOLVE, AHELP_CLOSEREASON_IC)
+			ticket.ICIssue(action_by_ckey)
+		if("close")
+			SSplexora.aticket_closed(ticket, action_by_ckey, AHELP_CLOSETYPE_CLOSE)
+			ticket.Close(action_by_ckey)
+		if("resolve")
+			SSplexora.aticket_closed(ticket, action_by_ckey, AHELP_CLOSETYPE_RESOLVE)
+			ticket.Resolve(action_by_ckey)
+		if("mhelp")
+			SSplexora.aticket_closed(ticket, action_by_ckey, AHELP_CLOSETYPE_CLOSE, AHELP_CLOSEREASON_MENTOR)
+			ticket.MHelpThis(action_by_ckey)
+
+/datum/world_topic/plx_sendaticketpm
+	keyword = "PLX_sendaticketpm"
+	require_comms_key = TRUE
+
+/datum/world_topic/plx_sendaticketpm/Run(list/input)
+  // We're kind of copying /proc/TgsPm here...
+	var/ticketid = text2num(input["id"])
+	var/input_ckey = input["ckey"]
+	var/sender = input["sender_ckey"]
+	var/stealth = input["stealh"]
+	var/message = input["message"]
+
+	var/requested_ckey = ckey(input_ckey)
+	var/ambiguious_target = disambiguate_client(requested_ckey)
+
+	var/client/recipient
+
+	if(istype(ambiguious_target, /client))
+		recipient = ambiguious_target
+
+	var/datum/admin_help/ticket
+	if (ticketid)
+		ticket = GLOB.ahelp_tickets.TicketByID(ticketid)
+	else
+		ticket = GLOB.ahelp_tickets.CKey2ActiveTicket(requested_ckey)
+
+	if (!ticket)
+		return "error=couldntfetchticket"
+
+	var/plx_tagged = "[sender](Plexora/External)"
+
+	var/adminname
+	var/stealthkey = sender
+	if(stealth)
+		adminname = "Administrator"
+	else
+		adminname = plx_tagged
+	stealthkey = GetTgsStealthKey()
```

Comment:
```suggestion
	var/adminname = stealth ? "Administrator" : plx_tagged
	var/stealthkey = GetTgsStealthKey()
```

Reply:
hm, i thought using tenarys was like, not good code practice